content/guides/full-text-search.md

```diff
@@ -0,0 +1,215 @@
+
+```sql
+SELECT
+    *
+  FROM documents
+  WHERE searchable @@ to_tsquery('english', 'flavorful');
+```
+
+Similarly, the following query returns the "PostgeSQL" row, even though the word "searching" does not appear in that row's text content, but "search" does.
+
+```sql
+SELECT
+    *
+  FROM documents
+  WHERE searchable @@ to_tsquery('english', 'searching');
+```
+
+The `@@` operator is a special operator which compares the `tsvector` value stored in the `searchable` column with the `tsquery` value provided in the query. The `tsquery` type is different from the `tsvector` type. For example, if you run the following command, Postgres will return the string "searching".
+
+```sql
+SELECT to_tsquery('english', 'searching');
+```
+
+## Search for multiple words
+
+If you try using `to_tsquery()` with multiple words, like `to_tsquery('english', 'searching text');`, Postgres will throw the following error.
+
+```
+ERROR: syntax error in tsquery: "searching text" (SQLSTATE 42601)
+```
+
+That's because the input to `to_tsquery()` must be tokens separated by `tsquery` operators like `&`. The correct way to search for "searching" and "text" would be `to_tsquery('english', 'searching & text');`. To make full text search easier to work with, Postgres also has a `phraseto_tsquery()` function that converts text into a `tsquery` with no need for operators. The following query will successfully return the "PostgreSQL" row.
+
+```sql
+SELECT
+    *
+  FROM documents
+  WHERE searchable @@ phraseto_tsquery('english', 'searching text');
+```
+
+`tsquery` also supports negations. For example, the following query will search for rows whose text content matches "searching" and does **not** match "text".
+
+```sql
+SELECT
+    *
+  FROM documents
+  WHERE searchable @@ to_tsquery('english', 'searching & !text');
+```
+
+Postgres also supports a `websearch_to_tsquery()` function, which uses an alternative syntax that doesn't require putting operators between all tokens. `websearch_to_tsquery()` supports negations by prefixing a token with `-`. The following query also searches for rows whose text content matches "searching" and does **not** match "text".
+
+```sql
+SELECT
+    *
+  FROM documents
+  WHERE searchable @@ websearch_to_tsquery('english', 'searching -text');
+```
+
+## Rank the results
+
+Postgres also provides two functions for ranking the results, allowing you to sort by which results are the best match. The following statement sorts rows that match "searching text" using the `ts_rank()` function, which counts the number of tokens that match.
+
+```sql
+SELECT
+    id,
+    title,
+    ts_rank(searchable, websearch_to_tsquery('english', 'searching text')) AS rank
+  FROM documents
+  WHERE searchable @@ websearch_to_tsquery('english', 'searching text')
+  ORDER BY rank DESC;
+```
+
+To see how sorting works in practice, insert two more rows as follows. The first row contains 6 tokens that match "search" and "text", so it should show up first.
```

Comment:
```suggestion
To see how sorting works in practice, insert two more rows as follows. The first row contains 6 tokens that match "search" and "text", so that row should show up first.
```